src/drivers/intel/cavs/ssp.c

```diff
@@ -365,12 +414,27 @@ static int ssp_set_config(struct dai *dai,
 		if (ssp_freq[clk_index].enc != CLOCK_SSP_XTAL_OSCILLATOR)
 			sscr0 |= SSCR0_ECS;
 	} else {
-		trace_ssp_error("ssp_set_config() error: BCLK %d",
-				config->ssp.bclk_rate);
-		ret = -EINVAL;
-		goto out;
-	}
+		/* check if we can get target BCLK with M/N */
+		for (i = MAX_SSP_FREQ_INDEX; i >= 0; i--) {
+			if (find_mn(ssp_freq[i].freq, config->ssp.bclk_rate,
+				    &mdiv, &i2s_m, &i2s_n))
+				clk_index = i;
```

Comment:
Should we break?

Reply:
In theory there may be higher clock that is not divisible and some lower that meets criteria.

Reply:
@tlauda If you mean break on *not found* then, it's situation like I wrote about, for *found* you may want to keep going because we prefer lower clocks and maybe there is lower clock that can be used.

Reply:
@jajanusz Should we optimize and start from the lower clock?

Reply:
@tlauda Yes, it will save a lot of processing in most cases.